parlai/core/torch_ranker_agent.py

```diff
@@ -91,7 +93,7 @@ def __init__(self, opt, shared=None):
             self.set_fixed_candidates(shared)
             self.set_vocab_candidates(shared)
 
-        self.rank_loss = nn.CrossEntropyLoss(reduce=True, size_average=False)
+        self.rank_loss = nn.CrossEntropyLoss(reduce=True, size_average=True)
```

Comment:
Undo this

Reply:
This is the most concerning, because without the size average, the (1) learning rate depends on the batch size and (2) we're more likely to get overflows.

This would explain why the learning rates I see in transformer rankers is always an order of magnitude lower than ones for transformer generators.

Reply:
I completely approve moving to average here (I did it for bert rankers for instance). But it means lr has to change?

Reply:
Yes, it does. @emilydinan  and @klshuster might have concerns on this, and I think it's a larger discussion.